### PR TITLE
MESH-1803/ Add missing tests for MESH-1792 & MESH-1767

### PR DESCRIPTION
**Pull request by JMoore96**
### modified logic

- adding tests for checking multisig errors and secondary key & permission limits

### CODE REVIEW

File: pallets/runtime/tests/src/multisig.rs [hunk 1]
```diff
@@ -101,6 +109,38 @@ fn join_multisig() {
...
+            vec![Signatory::from(alice_did), dave.signatory_acc()],
+            1,
+        ));
+
+        let dave_auth_id = get_last_auth_id(&dave.signatory_acc());
+        set_curr_did(Some(alice_did));
+        assert_noop!(
+            MultiSig::accept_multisig_signer_as_key(dave.origin(), dave_auth_id),
+            Error::SignerAlreadyLinkedToIdentity
+        );
```
Comment (Neopallium) [inline pallets/runtime/tests/src/multisig.rs:124]: Please add some comments to describe what is being tested.

File: pallets/multisig/src/lib.rs [hunk 2]
```diff
@@ -656,6 +656,8 @@ decl_error! {
         SignerAlreadyLinkedToMultisig,
         /// Signer is an account key that is already associated with an identity.
         SignerAlreadyLinkedToIdentity,
+        /// Multisig not allowed to add itself as a signer to itself.
+        MultisigNotAllowedToLinkToItself,
```
Comment (Neopallium) [inline pallets/multisig/src/lib.rs:660]:
```suggestion
        /// Multisig not allowed to add itself as a signer.
        MultisigNotAllowedToLinkToItself,
```

**Review by Neopallium — CHANGES_REQUESTED** [review]
Minor comment changes.  Otherwise looks good to me.

**Comment by Neopallium** [comment]
Also missing the tests for MESH-1767.

File: pallets/runtime/tests/src/identity_test.rs [hunk 3]
```diff
@@ -341,6 +341,63 @@ fn set_permission_to_signer_with_bad_perms() {
     });
 }
 
+#[test]
+fn ensure_perms_limited() {
```
[the remark this answers is not in the document]
Comment (JMoore96) [reply pallets/runtime/tests/src/identity_test.rs]: just finished making the change.

**Review by Neopallium — COMMENTED** [review]
The test for `Identity::cdd_register_did` looks good.  But we are still missing tests for the new errors from the MESH-1792 PR.  See the comment here: https://github.com/PolymathNetwork/Polymesh/pull/1226#discussion_r829192129

EDIT: Sorry, this is already done.  This PR looks good.

**Review by Neopallium — APPROVED** [review]

**Review by CJP10 — APPROVED** [review]

**Pull request merged**